src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -861,36 +862,117 @@ public bool EnC(byte[] meta, byte[] pdb)
             {
                 if (encMapHandle.Kind == HandleKind.MethodDebugInformation)
                 {
-                    var method = methods[asmMetadataReader.GetRowNumber(encMapHandle)];
-                    method.UpdateEnC(asmMetadataReaderParm, pdbMetadataReaderParm, i);
-                    i++;
+                    if (pdbMetadataReaderParm.GetRowNumber(encMapHandle) == number)
+                        return i;
                 }
+                i++;
             }
+            return -1;
         }
 
-        private void Populate()
+        internal string EnCGetString(StringHandle strHandle)
         {
-            var d2s = new Dictionary<int, SourceFile>();
+            var asmMetadataReaderLocal = asmMetadataReader;
+            var strIdx = strHandle.GetHashCode();
+            int i = 0;
+            while (strIdx > asmMetadataReaderLocal.GetHeapSize(HeapIndex.String))
+            {
+                strIdx = strIdx - asmMetadataReaderLocal.GetHeapSize(HeapIndex.String);
+                asmMetadataReaderLocal = enCMetadataReader[i];
+                if (strIdx < asmMetadataReaderLocal.GetHeapSize(HeapIndex.String))
+                    break;
+            }
+            return asmMetadataReaderLocal.GetString(MetadataTokens.StringHandle(strIdx));
+        }
 
-            SourceFile FindSource(DocumentHandle doc, int rowid, string documentName)
+        private void PopulateEnC(MetadataReader asmMetadataReaderParm, MetadataReader pdbMetadataReaderParm)
+        {
+            TypeInfo typeInfo = null;
+            int methodIdxAsm = 1;
+            foreach (var entry in asmMetadataReaderParm.GetEditAndContinueLogEntries())
             {
-                if (d2s.TryGetValue(rowid, out SourceFile source))
-                    return source;
+                if (entry.Operation == EditAndContinueOperation.AddMethod ||
+                    entry.Operation == EditAndContinueOperation.AddField)
+                {
+                    var typeHandle = (TypeDefinitionHandle)entry.Handle;
+                    if (!TypesByToken.TryGetValue(MetadataTokens.GetToken(asmMetadataReaderParm, typeHandle), out typeInfo))
+                    {
+                        int typeDefIdx = GetTypeDefIdx(asmMetadataReaderParm, asmMetadataReaderParm.GetRowNumber(entry.Handle));
+                        var typeDefinition = asmMetadataReaderParm.GetTypeDefinition(MetadataTokens.TypeDefinitionHandle(typeDefIdx));
+                        StringHandle name = MetadataTokens.StringHandle(typeDefinition.Name.GetHashCode() & 127);
+                        typeInfo = new TypeInfo(this, typeHandle, typeDefinition, asmMetadataReaderParm, true, logger);
+                        TypesByName[typeInfo.FullName] = typeInfo;
+                        TypesByToken[typeInfo.Token] = typeInfo;
+                    }
+                }
+                else
+                {
+                    if (entry.Handle.Kind == HandleKind.MethodDefinition)
```

Comment:
Just to make sure: will we get into this if also when `entry.Handle.Kind == HandleKind.MethodDebugInformation`? It's more or less the condition we had before, and the old code was preserved in this `if`. I'm not sure if EditAndContinueLogEntry's EntityHandle is not the equivalent of EntityHandle that was used previously, so shouldn't we add another condition:
`if (entry.Handle.Kind == HandleKind.MethodDefinition || entry.Handle.Kind == HandleKind.MethodDebugInformation`)?